XenAdmin/Wizards/ImportWizard/ImportSourcePage.cs

```diff
@@ -630,18 +632,15 @@ private void _unzipWorker_DoWork(object sender, DoWorkEventArgs e)
                         byte[] buffer = new byte[4 * 1024];
 
                         int bytesRead;
+                        ulong totalBytesRead = 0;
+
                         while ((bytesRead = bzis.Read(buffer, 0, buffer.Length)) > 0)
                         {
+                            totalBytesRead += (ulong)bytesRead;
                             outStream.Write(buffer, 0, bytesRead);
 
-                            int percentage = (int)Math.Floor((double)bzis.Position * 100 / length);
-
-                            if (percentage < 0)
-                                _unzipWorker.ReportProgress(0);
-                            else if (percentage > 100)
-                                _unzipWorker.ReportProgress(100);
-                            else
-                                _unzipWorker.ReportProgress(percentage);
+                            if (totalBytesRead % (128 * Util.BINARY_MEGA) == 0)

```

Comment:
This only reports progress if the total bytes hits a multiple of 128 MB (`134,217,728`) I think we should not report it like this. Our buffer is `4096` so let's maybe report based on that. Probably shouldn't be every `4096`, but a better multiple of it? If you were trying to avoid the bar flashing in and out immediately, I'd just use minimum size of bytes read before we use show the bar.

Also, we are likely never hitting 100%. I understand users might not see even see the progress bar at 100%, but this is just not accurate.

Reply:
Also I was playing with this and the original `.gz` file is removed. I understand we're telling the user but is it hard to not remove the original file? What is the idea behind doing so?